src/app/shell/item-comparators.ts

```diff
@@ -150,7 +150,21 @@ const ITEM_COMPARATORS: {
   // new -> old
   acquisitionRecency: acquisitionRecencyComparator,
   // None -> Kinetic -> Arc -> Thermal -> Void -> Raid -> Stasis
-  element: compareBy((item) => item.element?.enumValue ?? Number.MAX_SAFE_INTEGER),
+  elementWeapon: compareBy((item) => {
+    if (item.itemCategoryHashes.includes(ItemCategoryHashes.Weapon)) {
+      return item.element?.enumValue ?? Number.MAX_SAFE_INTEGER;
+    } else {
+      return undefined;
+    }
+  }),
+  // None -> Kinetic -> Arc -> Thermal -> Void -> Raid -> Stasis
```

Comment:
armor has a `DestinyEnergyTypeDefinition` not a `DestinyDamageTypeDefinition`
so its `enumValue` is a `DestinyEnergyType`
whose order is
  Any = 0,
  Arc = 1,
  Thermal = 2,
  Void = 3,
  Ghost = 4,
  Subclass = 5,
  Stasis = 6,